server/manager/src/main/java/org/apache/accumulo/manager/compaction/coordinator/CompactionCoordinator.java

```diff
@@ -429,11 +441,65 @@ public TNextCompactionJob getCompactionJob(TInfo tinfo, TCredentials credentials
       LOG.trace("No jobs found for group {}, returning empty job to compactor {}", groupName,
           compactorAddress);
       result = new TExternalCompactionJob();
+    } else {
+      LOG.info("Found job {}", result.externalCompactionId);
     }
 
     return new TNextCompactionJob(result, compactorCounts.get(groupName));
   }
 
+  protected CompletableFuture<TNextCompactionJob> getAsyncCompactionJob(TCredentials credentials,
+      String groupName, String compactorAddress, String externalCompactionId)
+      throws ThriftSecurityException {
+
+    // do not expect users to call this directly, expect compactors to call this method
+    if (!security.canPerformSystemActions(credentials)) {
+      throw new AccumuloSecurityException(credentials.getPrincipal(),
+          SecurityErrorCode.PERMISSION_DENIED).asThriftException();
+    }
+    CompactorGroupId groupId = CompactorGroupId.of(groupName);
+    LOG.trace("getCompactionJob called for group {} by compactor {}", groupId, compactorAddress);
+    TIME_COMPACTOR_LAST_CHECKED.put(groupId, System.currentTimeMillis());
+
+    return jobQueues.getAsync(groupId).thenApply(metaJob -> {
```

Comment:
Will eventually need to process these in another thread pool

Reply:
We should be able to just use `thenApplyAsync()` and pass in an executor. We could start with creating a regular thread pool for this but I'm wondering if this would be a time to look into Virtual threads